aspnet/fundamentals/error-handling.rst

```diff
@@ -1,8 +1,175 @@
+
+#. Once the headers for a response have been sent, you cannot change the response's status code, nor can any exception pages or handlers run. The response must be completed or the connection aborted.
+#. If the client disconnects mid-response, you cannot send them the rest of the content of that response.
+#. There is always the possibility of an exception occuring one layer below your exception handling layer.
+#. Don't forget, exception handling pages can have exceptions, too. It's often a good idea for production error pages to consist of purely static content.
+
+Following the above recommendations will help ensure your app remains responsive and is able to gracefully handle exceptions that may occur.
+
+Server Exception Handling
+-------------------------
+
+In addition to the exception handling logic in your app, the server hosting your app will perform some exception handling. If the server catches an exception before the headers have been sent it will send a 500 Internal Server Error response with no body. If it catches an exception after the headers have been sent it must close the connection. Requests that are not handled by your app will be handled by the server, and any exception that occurs will be handled by the server's exception handling. Any custom error pages or exception handling middleware or filters you have configured for your app will not affect this behavior.
+
+.. _startup-error-handling:
+
+Startup Exception Handling
+--------------------------
+
+One of the trickiest places to handle exceptions in your app is during its startup. Depending on how far into startup the app makes it before the exception occurs, exception handling middleware, filters, etc. may or may not be set up to deal with the problem. Exceptions that occur in your app's startup can also impact server behavior. For example, to enable SSL in Kestrel, one must configure the server with ``app.UseKestrelHttps()``. If an exception happens before this line in ``Startup``, then by default Kestrel will not handle the exception and the process will crash.
+
+HTTP 500 Errors on Azure
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your app throws an exception during startup, it is not handled by ``DeveloperExceptionPage`` (if configured). The app, if deployed to Azure (or another production IIS server), will return an HTTP 500 error with no message details to any request.
+
+The publish wizard in Visual Studio 2015 creates a *web.config* file if you don't have one. If you have a *web.config* file in the *wwwroot* folder, the deploy process inserts the markup into the *web.config* file it generates. 
+
+To get detailed error messages on Azure, add the following *web.config* file to the *wwwroot* folder.
+
+.. warning:: Security warning: Enabling detailed error message can leak critical information from your app. You should never enable detailed error messages on a production app.
+
+.. code-block:: html
+
+  <configuration>
+    <system.web>
```

Comment:
We will want a debugging azure doc or section, but not with any of the content currently in this section. Remove it and file a separate work item for azure docs.